File: app/lib/utils/departmentMapper.ts
export function getDepartmentFromRollNumber(rollNumber: string | null | undefined): string {
    if (!rollNumber) return 'Other';
    const upperRoll = rollNumber.toUpperCase().trim();

    if (upperRoll.includes('CS') || upperRoll.includes('COMPUTER')) return 'CSE';
    if (upperRoll.includes('ECE') || upperRoll.includes('ELECTRONICS') || upperRoll.includes('EC')) return 'ECE';
    if (upperRoll.includes('BT') || upperRoll.includes('BIO')) return 'BT';

    if (upperRoll.includes('AU') || upperRoll.includes('AUTO') || upperRoll.includes('MEA')) return 'MEA';
    if (upperRoll.includes('ME') || upperRoll.includes('MECHANICAL')) return 'ME';

    return 'Other';
}

/** Maps detected department to First Year group (A/B/C/D). */
export function getGroupFromDepartment(dept: string): string {
    switch (dept.toUpperCase()) {
        case 'CSE':
        case 'IT':  return 'A';
        case 'ECE':
        case 'EEE': return 'B';
        case 'ME':
        case 'MEA':
        case 'CIVIL': return 'C';
        case 'BT':
        case 'FT':  return 'D';
        default:    return 'A'; // fallback
    }
}

/** Converts a numeric profile year (1–4) to the year string stored in event_cal. */
export function getYearString(year: number | null | undefined): string {
    switch (year) {
        case 1: return '1st';
        case 2: return '2nd';
        case 3: return '3rd';
        case 4: return '4th';
        default: return 'General';
    }
}

File: mentron_flutter/lib/core/utils/department_mapper.dart
class DepartmentMapper {
  static String getDepartmentFromRoll(String? rollNumber) {
    if (rollNumber == null || rollNumber.isEmpty) return 'Other';
    final r = rollNumber.toUpperCase().trim();

    // MEA must come BEFORE ME to avoid false match
    if (r.contains('MEA') || r.contains('AU') || r.contains('AUTO')) return 'MEA';
    if (r.contains('ME') || r.contains('MECH') || r.contains('MECHANICAL')) return 'ME';
    if (r.contains('CSE') || r.contains('CS') || r.contains('COMPUTER')) return 'CSE';
    if (r.contains('ECE') || r.contains('EC') || r.contains('ELECTRONICS')) return 'ECE';
    if (r.contains('BT') || r.contains('BIO') || r.contains('BIOTECH')) return 'BT';

    return 'Other';
  }

  static const List<Map<String, String>> departments = [
    {'code': 'CSE', 'name': 'Computer Science & Engineering'},
    {'code': 'ECE', 'name': 'Electronics & Communication Engg'},
    {'code': 'ME',  'name': 'Mechanical Engineering'},
    {'code': 'MEA', 'name': 'Automobile Engineering'},
    {'code': 'BT',  'name': 'Bio Technology'},
  ];

  static String getName(String code) {
    return departments.firstWhere(
      (d) => d['code'] == code, 
      orElse: () => {'name': code}
    )['name']!;
  }

  static String getGroupFromDepartment(String? dept) {
    if (dept == null) return 'A'; // Fallback
    final d = dept.toUpperCase().trim();
    if (d == 'CSE') return 'A';
    if (d == 'ECE') return 'B';
    if (d == 'ME' || d == 'MEA') return 'C';
    if (d == 'BT') return 'D';
    return 'A'; // Fallback
  }
}
